### ISSUE TITLE
How to place an overlay on top of the image and customize the content？

### ISSUE THREAD

**Issue by FE-Friday** [OHIF/Viewers#2125]
hi, when I am developing an image application, I want to put an overlay on top of the image and customize the content, but I can’t find where to modify or configure it. Can you help me?

The picture below is what I want to set：

![微信图片_20201020164558](https://user-images.githubusercontent.com/58067499/96562804-c536fc00-12f3-11eb-99d3-1f6194341219.png)

**Comment by PoyangLiu** [comment]
You'll need to write a bit of code.  It's not as easy as changing some configuration file.

This PR https://github.com/OHIF/Viewers/pull/1831 for OHIF-203 shows a good blueprint on how to implement your own overlay.  

Basically you need to implement `getViewportModule` in your extension and provide a customized viewport object.
```
export const myOverlayExtension = {
    id: 'my-overlay-extension',

    getViewportModule(params) {
        return viewportModule(params);
    },
};
```

Then to tell the viewer to use your plugin, you can either

1. set the displayset.plugin to your own plugin name, otherwise it will default to the 'cornerstone' plugin.

    ```
    study.displaySets = studyMetadata.createDisplaySets(...);
    study.displaySets.forEach(ds => ds.plugin = 'my-overlay-extension');
    ```

2. setViewportSpecificData with { plugin: 'my-overlay-extension' } to override the default `cornerstone` plugin.

    ![image](https://user-images.githubusercontent.com/3383903/100123364-6b70a700-2e48-11eb-8108-a69cd99f9abb.png)

**Comment by PoyangLiu** [comment]
My previous post causes all sort of downstream effects for `toolModule` and `commandsModule`.

Another easier way, if you are building OHIF viewer, is to pass in your own [ViewportOverlay](https://github.com/cornerstonejs/react-cornerstone-viewport/tree/master/src/ViewportOverlay) class into the [`OHIFCornerstoneViewport`](https://github.com/OHIF/Viewers/blob/master/extensions/cornerstone/src/OHIFCornerstoneViewport.js#L235)

```
        <ConnectedCornerstoneViewport
             viewportOverlayComponent={MyViewportOverlayClass}
             ...
        ></ConnectedCornerstoneViewport
```

**Comment by ghost680** [comment]
I am very happy to receive the reply email.

It is great to implement this feature by configuring ConnectedCornerstoneViewport.

I have carefully read the source code of this place, and realized a custom overlay by configuring CornerstoneTool.

The OHIF/Viewer project is very good, thank you for your open source, let me solve a problem in my work.

Best wishes to you

发送自 Windows 10 版邮件应用

发件人: Jay
发送时间: 2020年11月25日 4:56
收件人: OHIF/Viewers
抄送: ghost; Author
主题: Re: [OHIF/Viewers] How to place an overlay on top of the image and customize the content？ (#2125)

My previous post causes all sort of downstream effects for toolModule and commandsModule.
Another easier way, if you are building OHIF viewer, is to pass in your own ViewportOverlay class into the OHIFCornerstoneViewport
        <ConnectedCornerstoneViewport
             viewportOverlayComponent={MyViewportOverlayClass}
             ...

**Comment by mokbelt** [comment]
Dears,
Trying to add Image overlay (from DICOM tags) to the view port of OHIF V3.
I tried to add the configuration mentioned in the customization service documentation (https://v3-docs.ohif.org/platform/services/ui/customization-service/), but still the name still didn't appear on the viewport.
Can someone exaplin how to pull tags (mostly Patient name/gender/dob etc...) to appear on the view port.
I built the V3 and serving it on the "dist" folder with nginx.

// in the APP_CONFIG file set the top right area to show the patient name
// using PN: as a prefix when the study has a non-empty patient name.
customizationService: {
  cornerstoneOverlayTopRight: {
    id: 'cornerstoneOverlayTopRight',
    customizationType: 'ohif.cornerstoneOverlay',
    items: [
      {
        id: 'PatientNameOverlay',
        // Note the ohif.overlayItem is a prototype instance for this object
        // The ohif.overlayItem is defined up above
        customizationType: 'ohif.overlayItem',
        attribute: 'PatientName',
        label: 'PN:',
      },
    ],
  },
},

![image](https://user-images.githubusercontent.com/56265911/219089492-087ba87e-0be6-4640-8442-89dd8ff39b95.png)

PS: In Developer tools it was mentioned, but didn't appear on the overlay:

![image](https://user-images.githubusercontent.com/56265911/219089661-40b74fea-3d9d-4e70-bb8f-05b3f4b07184.png)
